pyproject.toml

```diff
@@ -117,6 +116,9 @@ ignore = ["B008"]
 
 [tool.ruff.lint.per-file-ignores]
 "__init__.py" = ["F401"]
+# C901: warning for "too much branching logic"
+# in a function ...
+"ixmp4/data/api/base.py" = ["C901"]
```

Comment:
Would we really rather exclude this file here than make the function less complex? Usually, moving one or two if-clauses to a helper function is enough to remedy this warning.

Reply:
You're right, I removed that explicit exception and split up the response handling from that "complex" function. 
Kind of makes it harder to read imo but there you go!